File: THCS/2023_2024/Hue_2324_Phong/Bai2/BAI2.py
# Đọc dữ liệu từ file CHECK.INP
with open("CHECK.INP", "r") as infile:
    # Đọc số lượng hành khách
    N = int(infile.readline().strip())

    # Mở file để ghi kết quả
    with open("CHECK1.OUT", "w") as outfile1, open("CHECK2.OUT", "w") as outfile2:
        # Duyệt qua từng hành khách
        for i in range(1, N + 1):
            # Đọc các kiện hàng của hành khách
            line = infile.readline().strip()
            weights = list(map(float, line.split()))
            
            # Tính tổng trọng lượng và số lượng kiện hàng
            total_weight = sum(weights)
            item_count = len(weights)
            
            # Ghi tổng số cân nặng vào CHECK1.OUT, định dạng để không có .0 nếu là số nguyên
            if total_weight.is_integer():
                outfile1.write(f"{int(total_weight)}\n")
            else:
                outfile1.write(f"{total_weight}\n")
            
            # Kiểm tra điều kiện và ghi vào CHECK2.OUT nếu không đạt yêu cầu
            if total_weight > 20 or item_count > 10:
                outfile2.write(f"{i}\n")

File: THCS/2023_2024/Hue_2324_Phong/Bai2/BAI2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <sstream>
using namespace std;

int main() {
    ifstream infile("CHECK.INP");
    ofstream outfile1("CHECK1.OUT");
    ofstream outfile2("CHECK2.OUT");

    int N;
    infile >> N;
    infile.ignore();  // Bỏ qua ký tự newline sau số N

    for (int i = 1; i <= N; ++i) {
        string line;
        getline(infile, line);

        stringstream ss(line);
        vector<double> weights;
        double weight;
        double totalWeight = 0;
        int itemCount = 0;

        // Tính tổng trọng lượng và số lượng kiện hàng
        while (ss >> weight) {
            totalWeight += weight;
            itemCount++;
        }

        // Ghi tổng số cân nặng của hành khách vào file CHECK1.OUT
        outfile1 << totalWeight << endl;

        // Kiểm tra nếu hành khách không đủ điều kiện
        if (totalWeight > 20 || itemCount > 10) {
            outfile2 << i << endl;
        }
    }

    infile.close();
    outfile1.close();
    outfile2.close();
    return 0;
}
